packages/application-components/src/components/modal-page/modal-page.js

```diff
@@ -29,136 +28,74 @@ const parentSelector = () =>
-    ariaHideApp={false}
-    style={getStyles(props)}
-  >
-    <Spacers level={props.level} />
-    <div className={styles['page-container']}>
-      <Card className={styles['page-card']}>
-        <Spacings.Stack scale="m">
-          <div className={styles['page-header']}>
-            <Spacings.Inline
-              scale="m"
-              alignItems="center"
-              justifyContent="space-between"
-            >
-              <Text.Headline
-                elementType="h3"
-                truncate={true}
-                title={props.title}
-              >
-                {props.title}
-              </Text.Headline>
-              {props.onClose && (
-                <SecondaryIconButton
-                  label="Close page"
-                  onClick={props.onClose}
-                  icon={<CloseIcon size="medium" />}
-                />
-              )}
-            </Spacings.Inline>
-            <Spacings.Inline
-              scale="m"
-              alignItems="center"
-              justifyContent="space-between"
-            >
-              {props.subtitle}
-              <div>{props.components.actions}</div>
-            </Spacings.Inline>
-          </div>
-          <div className={styles['page-content']}>{props.children}</div>
-        </Spacings.Stack>
-      </Card>
-    </div>
-  </Modal>
+  <ClassNames>
+    {({ css: makeClassName }) => (
+      <Modal
+        isOpen={props.isOpen}
+        className={makeClassName(getContainerStyles(props))}
+        ariaHideApp={false}
+        contentLabel={props.title}
+        onRequestClose={props.onClose}
+        parentSelector={props.parentSelector || parentSelector}
+        overlayClassName={makeClassName(getOverlayStyles(props))}
+        shouldCloseOnEsc={Boolean(props.onClose)}
+        shouldCloseOnOverlayClick={Boolean(props.onClose)}
+      >
+        {props.showTopBar && (
+          <ModalPageTopBar
+            onClose={props.onClose}
+            currentPath={props.topBarLabels.currentPath || props.title}
+            previousPath={props.topBarLabels.previousPath}
+            tone={props.topBarTone}
+          />
+        )}
+
+        {props.showHeader && (
+          <ModalPageHeader
+            title={props.title}
+            actions={props.headerActions}
+            subtitle={props.subtitle}
+          />
+        )}
+        <div
+          css={css`
+            overflow: auto;
+            flex: 1;
+          `}
+        >
+          {props.children}
+        </div>
+      </Modal>
+    )}
+  </ClassNames>
 );
 ModalPage.displayName = 'ModalPage';
 ModalPage.propTypes = {
+  level: PropTypes.number,
+  title: PropTypes.string.isRequired,
+  zIndex: PropTypes.number,
   isOpen: PropTypes.bool.isRequired,
   onClose: PropTypes.func,
-  zIndex: PropTypes.number,
+  subtitle: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+  children: PropTypes.node.isRequired,
+  showTopBar: PropTypes.bool,
+  showHeader: PropTypes.bool,
```

Comment:
Have you considered having different components with specific properties instead of one component with many different combinations?
For example, the dialog components have been semantically split into Info,Confirmation,Form components, even though their APIs are very similar as they are built with the same set of internal components. The reason for splitting them is to give them semantic names, which makes it clear when to use what and to keep the respective APIs restrictive in regards to the unique properties of each components.

With that in mind, I would expect something similar to the “modal page” components, for example:
- `InfoModalPage` (the one without actions)
- `FormModalPage` (the one with actions and white header background, without any sub pages)
- `TabularModalPage` (the one with actions and gray header background, with sub pages as tabs)

The names can change, but the idea is what I’m trying to explain.

Reply:
Yes, I've considered that, even though I first opted for this more "generic" version of this Modal Page, but I've already separated some of the elements into internal components (topBar and Header) with that in mind.

I only left out the Tabular navigation: currently the dev would pass that whole part as the content (children) of the modal, and "turn off" the header.

But if we all agree on this, I can build on top of the current implementation and provide the 3 different, more specific versions of it 🙂

Reply:
Yes, for me it comes down to make the lives of the developers easier, by reducing the cognitive overhead (in this case in terms of correct combinations of props based on the type of usage).

Reply:
Regarding the `TabularModalPage`: If I understand correctly, if we're gonna integrate the tab-page navigation in this ModalPage, we're gonna need to migrate the Tab components from the MC-FE to app-kit as well.

In fact, I think having a Tab/Tabs component is actually agnostic enough to be a good candidate for UI-kit! What do you think?

Reply:
I also think Tab component would fit for UI Kit.

Reply:
To clarify: I didn’t mean to integrate tabs in this component, it’s just a name to suggest the usage of item regards to its layout.
For instance, the FormDialog also does not implement a form, it just indicates that it should be used for form dialogs

Reply:
Ah, I see, so in this case, the developer would use the "premade" version of `TabularModalPage` which "premade" with the rest of the pieces of the Modal such as TopBar and Header already styled accordingly, and would then nest the Tab navigation and TabContent within the `children` of the modal.

Reply:
Yes, the tabs part is left to the children.